cassandra-4/src/main/java/io/debezium/connector/cassandra/Cassandra4CommitLogProcessor.java

```diff
@@ -96,16 +94,18 @@ public void stop() {
         super.stop();
     }
 
-    final static Set<Pair<CommitLogProcessingCallable, Future<ProcessingResult>>> submittedProcessings = ConcurrentHashMap.newKeySet();
-
     private void submit(Path index) {
-        CommitLogProcessingCallable callable = new CommitLogProcessingCallable(new LogicalCommitLog(index.toFile()),
-                queues,
-                metrics,
-                Cassandra4CommitLogProcessor.this.context);
+        final Cassandra4CommitLogParserBase cassandra4CommitLogParserBase;
 
-        Future<ProcessingResult> future = executorService.submit(callable);
-        submittedProcessings.add(new Pair<>(callable, future));
+        if (context.getCassandraConnectorConfig().isCommitLogRealTimeProcessingEnabled()) {
+            cassandra4CommitLogParserBase = new Cassandra4CommitLogNearRealTimeParser(new LogicalCommitLog(index.toFile()), queues, metrics, this.context);
```

Comment:
you could simplify this, both constructors are taking same variables in.

Reply:
Sorry, I did not understand here. Do you want to create a method in abstract class and set member variables at a common place, and that setter is called outside of if/else block?